src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -79,40 +99,275 @@ static DamageTable D_8099F4E8 = {
+void EnZo_FollowPath(EnZo* this, GlobalContext* globalCtx) {
+    s16 speed;
+    Vec3f pos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099E9E0.s")
+    Math_SmoothStepToF(&this->actor.speedXZ, 1.0f, 0.4f, 1000.0f, 0.0f);
+    speed = this->actor.speedXZ * 400.0f;
+    if (func_8013D68C(this->path, this->waypoint, &pos) && func_8013D768(&this->actor, &pos, speed)) {
+        this->waypoint++;
+        if (this->waypoint >= this->path->count) {
+            this->waypoint = 0;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EA60.s")
+    if (this->actor.depthInWater > 60.0f) {
+        EnZo_SetAnimation(&this->skelAnime, 1);
+        this->actionFunc = EnZo_TreadWater;
+        this->actor.gravity = 0.0f;
+        this->actor.speedXZ = 0.0f;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EBD8.s")
+void EnZo_TreadWater(EnZo* this, GlobalContext* globalCtx) {
+    f32 targetYVel;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EC50.s")
+    if (this->actor.depthInWater < (sREG(0) + 50.0f)) {
+        targetYVel = -1.0f;
+    } else {
+        targetYVel = 1.0f;
+    }
+    Math_ApproachF(&this->actor.velocity.y, targetYVel, (sREG(1) + 18.0f) * 0.01f, (sREG(2) + 12.0f) * 0.01f);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099ED4C.s")
+void EnZo_DoNothing(EnZo* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EE24.s")
+void EnZo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+    s32 pad;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600D208, NULL, this->jointTable, this->morphTable, 20);
+    EnZo_SetAnimation(&this->skelAnime, 0);
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+
+    this->path = func_8013D648(globalCtx, ENZO_GET_PATH(&this->actor), ENZO_NO_PATH);
+    Actor_SetScale(&this->actor, 0.01f);
+
+    this->actionFunc = EnZo_Walk;
+    this->actor.gravity = -4.0f;
+}
+
+void EnZo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnZo_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    EnZo_LookAtPlayer(this, globalCtx);
+    EnZo_PlayWalkingSound(this, globalCtx);
+    EnZo_UpdateCollider(this, globalCtx);
+}
+
+s32 EnZo_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* actor,
+                          Gfx** gfx) {
+    EnZo* this = (EnZo*)actor;
```

Comment:
```suggestion
s32 EnZo_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx,
                          Gfx** gfx) {
    EnZo* this = THIS;
```
If we use `thisx` for the actor, then we can use the `THIS` macro.